ts/src/test/Exchange/base/test.market.ts

```diff
@@ -102,23 +102,53 @@ function testMarket (exchange, skippedProperties, method, market) {
     const contractSize = exchange.safeString (market, 'contractSize');
     // contract fields
     if (market['contract']) {
+        // spot should be false
+        assert (!market['spot'], '"spot" must be false when "contract" is true' + logText);
+        // todo: expand logic on other market types, like "options", "index" etc.
+        // assert (isSwapOrFuture, 'either swap or future needs to be true when "contract" is true' + logText);
         // linear & inverse should have different values (true/false)
-        // todo: expand logic on other market types
         if (isSwapOrFuture) {
-            assert (market['linear'] !== market['inverse'], 'market linear and inverse must not be the same' + logText);
-            if (!('contractSize' in skippedProperties)) {
-                // contract size should be defined
-                assert (contractSize !== undefined, '"contractSize" must be defined when "contract" is true' + logText);
-                // contract size should be above zero
-                assert (Precise.stringGt (contractSize, '0'), '"contractSize" must be > 0 when "contract" is true' + logText);
-            }
-            if (!('settle' in skippedProperties)) {
-                // settle should be defined
-                assert ((market['settle'] !== undefined) && (market['settleId'] !== undefined), '"settle" & "settleId" must be defined when "contract" is true' + logText);
+            if (!('subType' in skippedProperties)) {
```

Comment:
add an ability to be skipped for specific exchange temporarily. 
we will eventually move many skips after we have  revised implementations